syntax = "proto3";

// 로그인시 클라이언트에서 서버로 전송하는 패킷이다.
message LoginRequest {
	string id = 1;		// 유저의 id
	string passwd = 2;  // 유저의 pw
}

// 로그인 요청 처리에 따른 서버의 응답 패킷이다.
message LoginResponse {
	int32 response_code = 1;  // 처리에 대한 정보를 담고 있음
}

// 채널 서버에서 유저 세션을 생성한 후 클라이언트로 보내주는 패킷이다.
message UserInfo {
	int32 identify_id = 1;
	int32 channel_num = 2;
	int32 session_num = 3;
}

// 동기화를 위해 서버에서 유저들의 정보들을 뿌려주기 위해 사용하는 패킷이다.
message SyncInfo_S {
	message User {
		int32 user_id = 1;
		int32 x_pos = 2;
		int32 y_pos = 3;
		int32 vec = 4;
		int32 type = 5;
		int32 A_type_pow = 6;
		int32 B_type_pow = 7;
		int32 C_type_pow = 8;
	}
	repeated User users = 1;
}

// 동기화를 위해 클라이언트에서 서버로 주기적으로 보내주는 패킷이다.
message SyncInfo_C {
	int32 user_id = 1;
	int32 x_pos = 2;
	int32 y_pos = 3;
	int32 vec = 4;
	int32 type = 5;
	int32 A_type_pow = 6;
	int32 B_type_pow = 7;
	int32 C_type_pow = 8; 
}